pyproject.toml

```diff
@@ -78,3 +67,13 @@ filterwarnings = [
     "ignore:Attribute s is deprecated and will be removed in Python 3.14:DeprecationWarning",
     "ignore:ast.NameConstant is deprecated and will be removed in Python 3.14:DeprecationWarning",
 ]
+
+### ======================================================================== ###
+
+[tool.ruff.lint]
+select = ["E", "F", "B", "I"]
+ignore = ["B018", "B904"]
```

Comment:
Both of these seems relatively useful, but I can see B904 being quite noisy on the current codebase. I'll re-enable them in a separate PR & clean up the code

Reply:
B018: https://docs.astral.sh/ruff/rules/useless-expression/

B904: https://docs.astral.sh/ruff/rules/raise-without-from-inside-except/